pkg/backend/diy/backend.go

```diff
@@ -1334,6 +1338,26 @@ func (b *diyBackend) UpdateStackTags(ctx context.Context,
 	return errors.New("stack tags not supported in diy mode")
 }
 
+// UpdateStackDeployment updates the stacks's deployment settings.
+func (b *diyBackend) UpdateStackDeployment(ctx context.Context, stack backend.Stack,
+	deployment apitype.DeploymentSettings,
+) error {
+	// The local backend does not currently persist tags.
+	return errors.New("stack deployments not supported in --local mode")
```

Comment:
```suggestion
	return errors.New("stack deployments not supported with diy backends")
```

Similar below, as someone might not be using `--local`, but azure blob or s3 or similar backends, where `--local` would be somewhat confusing.